src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogLogRenderer.cs

```diff
@@ -40,12 +43,40 @@ public string RenderDatadogEvent(LogEvent logEvent)
             _formatter.Format(logEvent, payloadWriter);
             var rawPayload = payloadWriter.ToString();
 
-            var rawLogSize = Encoding.UTF8.GetByteCount(rawPayload);
-            if (rawLogSize > _maxMessageSize)
+            var truncated = TruncateIfNeeded(rawPayload);
+            return truncated.Select(x => ToDDPayload(Encoding.UTF8.GetString(x))).ToArray();
+        }
+
+        internal IEnumerable<byte[]> TruncateIfNeeded(string rawPayload) {
+            
+            var bytes = Encoding.UTF8.GetBytes(rawPayload);
+
+            var grouped = Enumerable.Range(0, (bytes.Count() / _maxMessageSize) + 1)
+                .Select((b, i) => bytes.Skip(i * _maxMessageSize)
+                                       .Take(_maxMessageSize))
```

Comment:
After inspecting the agent code - this (the way ive implemented it) is how the agent handles it (not considering the increase in message size). 

However I think we probably should consider reserving space since we are now allowed to get close to the true message size limit. Previously, overflowing by a few bytes would have no impact since the intake could handle the larger payload. 

Another thing We should consider/test is: Does the outer (dd payload) impact the max message size? I think it does...
TBH It's hard to account for all of these cases when truncated the payload - we should probably advise users to set the limit somewhat under 1MB (`900K`) for example for safety.

Reply:
Ok I added support to account for both the size of the `...TRUNCATED...` flag as well as the size of the parent DD payload. This ensure that it is impossible to overflow the payload size past the user defined limit. 

We should also do this in the agent when we visit these changes there.